test/unit/clarity_ext/dilution/test_dilutions.py

```diff
@@ -0,0 +1,74 @@
+import unittest
+from mock import MagicMock
+from clarity_ext.dilution import DilutionScheme
+from test.unit.clarity_ext.helpers import fake_analyte
+
+
+class TestDilutionScheme(unittest.TestCase):
+    """
+    Validates that the DilutionScheme property of the context object returns the expected values
+    based on fake data from the LIMS
+    """
+
+    def test_dilution_scheme_hamilton_expected_results(self):
+        """Dilution scheme created by mocked analytes is correctly generated for Hamilton"""
+        # Setup:
+        repo = MagicMock()
+        repo.all_analytes = all_analyte_test_set1
+        dilution_scheme = DilutionScheme(repo, "Hamilton")
+
+        expected = [
+            ['art-name1', '36', 'DNA1', '14.9', '5.1', '34', 'END1'],
+            ['art-name2', '33', 'DNA2', '14.9', '5.1', '17', 'END2'],
+            ['art-name3', '50', 'DNA2', '14.9', '5.1', '44', 'END1'],
+            ['art-name4', '93', 'DNA2', '14.9', '5.1', '69', 'END2']]
+
+        # Test:
+        actual = [
+            [dilute.sample_name,
+             "{}".format(dilute.source_well_index),
+             dilute.source_plate_pos,
+             "{:.1f}".format(dilute.sample_volume),
+             "{:.1f}".format(dilute.buffer_volume),
+             "{}".format(dilute.target_well_index),
+             dilute.target_plate_pos] for dilute in dilution_scheme.dilutes
+        ]
+
+        # Assert:
+        self.assertEqual(expected, actual)
+
+
+def all_analyte_test_set1():
+    """
+    Returns a list of (inputs, outputs) fake analytes for a particular step.
+
+    Analytes have been sorted, as they would be when queried from the repository.
+    """
+    inputs = [
+        fake_analyte("cont-id1", "art-id1", "sample1", "art-name1", "D:5", concentration=134),
```

Comment:
It would be cool to have tests for the StepRepository, I think, but I think it's important to keep it super simple, only a wrapper around the REST proxy. I might be misunderstanding though.